src/find.ts

```diff
@@ -13,10 +13,10 @@ export type { Options };
  * @returns The absolute path to the item, if found.
  */
 export function up(name: string, options?: Options): string | undefined {
-	let dir: string, tmp: string;
-	let start = options && options.cwd || '';
-	for (dir of walk.up(start, options)) {
-		tmp = join(dir, name);
+	const dirs = walkUp(options?.cwd ?? '', options);
+
+	for (let i = 0, l = dirs.length; i < l; i++) {
+		const tmp = join(dirs[i], name);
```

Comment:
I know about generational GC, but it doesn't win the trade off here. In JS loops, the cost of declaring variables _inside_ a loop outweighs the GC signaling.... cuz it means the GC is doing more work to cleanup. 

That said, the dirs coming from walk.up are realistically 3-20 parents long. Do either of these factors make much difference then? No. Had this been a significantly larger array to loop over, then we'd see a bigger difference with variable _reuse_ coming out on top. I see it time and time again.

Btw, the `const` performance benefits is a fairy tale. In fact, for the longest time (maybe even still, but I dont think so & hope not) consts were technically longer to initialize. They're practically decorative signals to the reader.

Reply:
As mentioned before, the change in for-of to a traditional for loop here is the only significant performance-related change I've seen (so far).

Reply:
Finished the detailed review, can confirm that this was the only significant/measurable/worthwhile performance-related change.

Reply:
I actually did some tests and declaring vars in loops are faster
Can you do some performance tests to check this?

Reply:
I've been doing these tests for over a decade & find it to be true every time I release a library that has loops.

Reply:
Please post your benchmark here
Or pls dm me on discord: r.e.v.e

Reply:
I have already spent over an hour and half on this PR review, which is considerably more than I’ve done in the past. 

All you’ve done is reference performance principles I already know and try to blanket-apply them across the board. I’ve given you plenty of validation re the ideas, other tips, and stated the trade-offs but you don’t seem to be interested. 

I’ve written code like this/you for quite a while. It’s not always “best” and you have to take the larger /usage context of what it is you’re writing/optimizing in mind. 

While still not much code overall, and without checking explicitly, all these accounts probably add 200-300 bytes overall? Which again, isn’t much, but it’s nearly +50-150% of the entire package size for (according to your numbers, and im guessing) ~1000 op/sec on a benchmark where it’s *already* 123k to 5k. 

(And that’s trusting you even see a performance increase. I see none with this PR on my machines)

Thank you for the PR and interest, but the numbers don’t lie. Some of what you’ve suggested would/does certainly apply at larger scales. But here, they don’t. And that’s ultimately because we are dealing with smaller arrays AND because ultimately it’s bottlenecked by filesystem.